### PR TITLE
Make bundle compatible with Sulu 2.0

### PR DESCRIPTION
**Pull request by niklasnatter**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

This PR refactors the bundle to make it compatible with the newly release version 2.0 of Sulu.

### CODE REVIEW

**Comment by nnatter** [comment]
remove styleci config

**Comment by nnatter** [comment]
rename upgrade file

**Comment by nnatter** [comment]
adjust travis phpunit call

File: UPGRADE.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 ## dev-develop
 
+### Rename comment_count property to commentCount
+
+The `comment_count` property of serialized threads was renamed to `commentCount`.
+
+
```
Comment (alexander-schranz) [inline UPGRADE.md]: remove line

File: Manager/CommentManager.php [hunk 2]
```diff
@@ -72,10 +72,10 @@ public function addComment(
             $thread->setTitle($threadTitle);
         }
 
-        $this->dispatcher->dispatch(Events::PRE_PERSIST_EVENT, new CommentEvent($type, $entityId, $comment, $thread));
+        $this->dispatcher->dispatch(new CommentEvent($type, $entityId, $comment, $thread), Events::PRE_PERSIST_EVENT);
```
Comment (alexander-schranz) [inline Manager/CommentManager.php:75]: We should refractor the Events to use for Every Event an own class.
E.g. `CommentPersistEvent`

Like in https://symfony.com/blog/new-in-symfony-4-3-simpler-event-dispatching

File: README.md [hunk 3]
```diff
@@ -20,8 +20,8 @@ and code are not stable yet (pre 1.0).
 ## Requirements
 
 * Composer
-* PHP `^7.1`
-* Sulu `^2.0.0-RC1`
+* PHP `^7.2`
+* Sulu `^2.0.0`
```
Comment (alexander-schranz) [inline README.md]: would just write `^2.0`

File: Resources/config/routing_api.yml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-    resource: Sulu\Bundle\CommentBundle\Controller\CommentController
+    resource: sulu_comment.comment_controller
     options:
         expose: true
 
 sulu_comment.threads:
     type: rest
     name_prefix: sulu_comment.
-    resource: Sulu\Bundle\CommentBundle\Controller\ThreadController
+    resource: sulu_comment.thread_controller
```
Comment (alexander-schranz) [inline Resources/config/routing_api.yml:11]: 👍

File: UPGRADE.md [hunk 5]
```diff
@@ -2,6 +2,11 @@
 
 ## dev-develop
 
+### Rename comment_count property to commentCount
+
+The `comment_count` property of serialized threads was renamed to `commentCount`.
```
Comment (alexander-schranz) [inline UPGRADE.md:7]: this should not change can we check the jms serializer config in the bundle:

Sulu overwrite it here:

https://github.com/sulu/sulu/blob/87fdd3500d5999548acc07ca4b4df9842a9956f4/src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php#L129-L138

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
The eventDispatcher upgrade can be seen as optional.

File: Manager/CommentManager.php (hunk 2 above)
Comment (niklasnatter) [reply Manager/CommentManager.php:75]: you are right 🙂 but this does not trigger a deprecation-error right now, so i think it is okay to not refactor this right now

File: UPGRADE.md (hunk 5 above)
Comment (niklasnatter) [reply UPGRADE.md:7]: looks like the bundle does not prepend any `property_naming` config. how should we deal with this?

Comment (niklasnatter) [reply UPGRADE.md:7]: in general i would prefer `commentCount` to `comment_count`.

Comment (alexander-schranz) [reply UPGRADE.md:7]: The bundle should not need todo it was the sulu/sulu does it.
Its  not about whats better its just a bc break which we don't know where it comes from and that was worried me.

File: Manager/CommentManager.php (hunk 2 above)
Comment (alexander-schranz) [reply Manager/CommentManager.php:75]: You are right!

**Pull request merged**